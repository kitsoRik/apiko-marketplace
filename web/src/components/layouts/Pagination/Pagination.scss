.pagination {
    padding: 5px;

    display: flex;
    justify-content: space-between;
    align-items: center;

    border: 2px solid #349A89;
}