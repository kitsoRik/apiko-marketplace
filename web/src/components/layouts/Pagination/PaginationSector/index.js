import PaginationSector from "./PaginationSector";

export default PaginationSector;