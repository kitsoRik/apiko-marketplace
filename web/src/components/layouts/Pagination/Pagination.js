import React from 'react';

import "./Pagination.scss";
import Form from '../../layouts/Form';
import PaginationSector from './PaginationSector/PaginationSector';

import _ from 'lodash';

const MAX_PAGE_SEQUENCE = 3;

const Pagination = ({ className, pages = 0, page = 16, onChangePage = (page) => { } }) => {

    const middlePages = _.times(MAX_PAGE_SEQUENCE + 2, p => page + p - 2)
        .filter(p => (p > 0 && p <= pages));

    const leftPages = _.times(MAX_PAGE_SEQUENCE, p => p + 1)
        .filter((p, i) => i < 5 && (p > 0 && p <= pages && !middlePages.find(m => m < p || m === p)));
    const rightPages = _.times(MAX_PAGE_SEQUENCE, p => pages - 5 + p + 3)
        .filter((p, i) => i < 5 && (p > 0 && p <= pages && !middlePages.find(m => m > p || m === p)));


    const changePage = (p) => {
        if (p === page) return;
        onChangePage(p);
    }

    return (
        <Form className={`pagination ${className ?? ""}`}>
            <PaginationSector disabled={page === 1} className="pagination-sector-prev" value="<" onClick={() => changePage(page - 1)} />
            {leftPages?.map(p => <PaginationSector key={p} isCurrent={page === p} value={p} onClick={() => changePage(p)} />)}
            {leftPages.length !== 0 && <span>...</span>}

            {middlePages.map(p => <PaginationSector key={p} isCurrent={page === p} value={p} onClick={() => changePage(p)} />)}

            {rightPages.length !== 0 && <span>...</span>}
            {rightPages?.map(p => <PaginationSector key={p} isCurrent={page === p} value={p} onClick={() => changePage(p)} />)}
            <PaginationSector disabled={page >= pages} className="pagination-sector-next" value=">" onClick={() => changePage(page + 1)} />
        </Form>
    )
};

export default Pagination;