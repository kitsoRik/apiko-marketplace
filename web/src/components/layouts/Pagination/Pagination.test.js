import React from "react";
import { create } from "react-test-renderer";
import Pagination from "./Pagination";

describe("Pagination", () => {
	it("should be rendered", () => {
		const component = create(<Pagination />);
		const json = component.toJSON();

		expect(json).toMatchSnapshot();
	});
	it("should be rendered", () => {
		const component = create(<Pagination page={1} pages={10} />);
		const json = component.toJSON();

		expect(json).toMatchSnapshot();
	});
});
