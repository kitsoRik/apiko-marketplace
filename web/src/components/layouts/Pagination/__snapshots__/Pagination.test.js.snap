// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination should be rendered 1`] = `
<div
  className="form pagination "
>
  <button
    className="pagination-sector pagination-sector-prev"
    current={null}
    disabled={null}
    onClick={[Function]}
  >
    &lt;
  </button>
  <button
    className="pagination-sector pagination-sector-next"
    current={null}
    disabled={true}
    onClick={[Function]}
  >
    &gt;
  </button>
</div>
`;

exports[`Pagination should be rendered 2`] = `
<div
  className="form pagination "
>
  <button
    className="pagination-sector pagination-sector-prev"
    current={null}
    disabled={true}
    onClick={[Function]}
  >
    &lt;
  </button>
  <button
    className="pagination-sector "
    current="true"
    disabled={null}
    onClick={[Function]}
  >
    1
  </button>
  <button
    className="pagination-sector "
    current={null}
    disabled={null}
    onClick={[Function]}
  >
    2
  </button>
  <button
    className="pagination-sector "
    current={null}
    disabled={null}
    onClick={[Function]}
  >
    3
  </button>
  <span>
    ...
  </span>
  <button
    className="pagination-sector "
    current={null}
    disabled={null}
    onClick={[Function]}
  >
    8
  </button>
  <button
    className="pagination-sector "
    current={null}
    disabled={null}
    onClick={[Function]}
  >
    9
  </button>
  <button
    className="pagination-sector "
    current={null}
    disabled={null}
    onClick={[Function]}
  >
    10
  </button>
  <button
    className="pagination-sector pagination-sector-next"
    current={null}
    disabled={null}
    onClick={[Function]}
  >
    &gt;
  </button>
</div>
`;
